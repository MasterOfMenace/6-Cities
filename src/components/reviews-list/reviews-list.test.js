import React from 'react';
import renderer from 'react-test-renderer';
import ReviewsList from './reviews-list.jsx';

const Dates = [
  `10 July 2019`,
  `25 April 2018`
];

const mockReviews = [
  {
    author: `author-1`,
    avatar: `author-avatar-1`,
    text: `Review text`,
    time: new Date(Dates[0]).toISOString()
  },
  {
    author: `author-2`,
    avatar: `author-avatar-2`,
    text: `Review text`,
    time: new Date(Dates[1]).toISOString()
  }
];

it(`Правильное отображение компонента ReviewsList`, () => {
  const tree = renderer
    .create(<ReviewsList
      reviews={mockReviews}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
