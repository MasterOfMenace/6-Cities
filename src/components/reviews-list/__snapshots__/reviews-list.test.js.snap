// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Правильное отображение компонента ReviewsList 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="author-avatar-1"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        author-1
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Review text 1
      </p>
      <time
        className="reviews__time"
        dateTime={2019-07-10T00:00:00.000Z}
      >
        July 2019
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="author-avatar-2"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        author-2
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Review text 2
      </p>
      <time
        className="reviews__time"
        dateTime={2018-04-25T00:00:00.000Z}
      >
        April 2018
      </time>
    </div>
  </li>
</ul>
`;
